diff --git a/d/sphinxcontrib-newsfeed-0.1.4/sphinxcontrib/newsfeed.py b/exts/newsfeed.py
index 57b0451..62c2345 100644
--- a/d/sphinxcontrib-newsfeed-0.1.4/sphinxcontrib/newsfeed.py
+++ b/exts/newsfeed.py
@@ -26,8 +26,26 @@ class FeedDirective(Directive):
         output = []
         entries = []
         includefiles = []
-        for entry in self.content:
-            if not entry:
+
+        def expand_content(content):
+            ''' Support globbing, so we don't need to enumerate each page.
+            '''
+            import fnmatch, re
+            docs = None
+            for entry in content:
+                if "*" in entry:
+                    if docs is None:
+                        docs = sorted(env.found_docs)
+                    re_comp = re.compile(fnmatch.translate(entry))
+                    # reversed since this is a newsfeed!
+                    for e in reversed(docs):
+                        if re_comp.match(e):
+                            yield e
+                else:
+                    yield entry
+
+        for entry in expand_content(self.content):
+            if not entry or entry == env.docname:
                 continue
             docname = docname_join(env.docname, entry)
             if docname not in env.found_docs:
